cincinnati/src/plugins/internal/channel_filter.rs

```diff
@@ -0,0 +1,329 @@
+//! This plugin can be used to filter a graph by a specific ! channel.
+//! It reads the requested channel from the parameters value at key ! "channel",
+//! and the value must match the regex specified at CHANNEL_VALIDATION_REGEX_STR
+
+use failure::Fallible;
+use plugins::{InternalIO, InternalPlugin};
+
+pub struct ChannelFilterPlugin {
+    pub key_prefix: String,
+    pub key_suffix: String,
+}
+
+static CHANNEL_VALIDATION_REGEX_STR: &str = r"[0-9a-zA-Z]+";
```

Comment:
Do we have a specification for these or did you just come up with some reasonable values?

Reply:
I came up with these. I adapted them again to match the channels we currently envision.